### ISSUE TITLE
Add benchmarks to vergesort on top of several sorting algorithms

### ISSUE THREAD

**Issue by Morwenn** [Morwenn/vergesort#8]
Since vergesort is more of an additional layer on top of another sorting algorithm, it would be great to show how it performs when other several sorting algorithms are used as a fallback. This would show its actual power as an « optimization layer » rather than what it's worth as a stand-alone sorting algorithm.

To make it look good, the trick would be to provide graphs to show how it performs when using well-known and performant algorithms as a fallback:
* `std::sort`
* `std::stable_sort`
* timsort
* pdqsort
* spreadsort
* ska_sort

On the other hand, I wouldn't like to lie to make it look better than it is, so I should make efforts to create distributions where vergesort makes things worse:
* Ascending/descending sawtooth patterns with runs slightly smaller than `n / log n`.
* Descending runs of size `n / log n` (to force use of `std::reverse` inteleaved with pipe-organ runs of size `n / log n` too to force vergesort to waste time scanning, and to trigger one of pdqsort's not-so-good patterns as well.

**Comment by Morwenn** [comment]
So... I tried to implement the distributions mentioned above to make sure that vergesort would hit its worst cases, maximizing the number of comparisons needed before it fails, and then maximizing the number of runs it has to merge, trying to make it fallback to one of pdqsort's worst cases every time. Basically the goal was to try to make it perform a maximum of unneeded extra work before falling back to pdqsort, but...

![I tried to kill vergesort once...](https://i.imgur.com/5tod3KH.png)

For the ascending and descending sawtooth distributions, where runs are slightly smaller than n / log n, I *guess* that I actually have branch prediction on my side as long as the run is sorted or reverse-sorted, so the extra time to check for presortedness isn't even noticeable.

For the « Vergesort killer » distribution, despite my efforts to make it perform as much unneeded work as possible, taking advantage of the already reverse-sorted runs seems to totally outweight the cost of using several times pdqsort and using an as-big-as-possible k-way merge at the end of the algorithm. To the point that vergesort still performs better than pdqsort for a distribution that was supposed to be one of its worst cases...

---

Now it's still integers we're talking about. There are chances that it behaves differently for different data types.
